components/infobox/wikis/dota2game/infobox_cosmetic_custom.lua

```diff
@@ -0,0 +1,207 @@
+---
+-- @Liquipedia
+-- wiki=dota2game
+-- page=Module:Infobox/Cosmetic/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local CosmeticIcon = require('Module:Cosmetic')
+local DateExt = require('Module:Date/Ext')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Template = require('Module:Template')
+
+local Injector = Lua.import('Module:Infobox/Widget/Injector')
+local Cosmetic = Lua.import('Module:Infobox/Cosmetic')
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Builder = Widgets.Builder
+local Cell = Widgets.Cell
+local Center = Widgets.Center
+local Title = Widgets.Title
+
+---@class Dota2CosmeticInfobox: CosmeticInfobox
+local CustomCosmetic = Class.new(Cosmetic)
+local CustomInjector = Class.new(Injector)
+
+---@param frame Frame
+---@return Html
+function CustomCosmetic.run(frame)
+	local cosmetic = CustomCosmetic(frame)
+	cosmetic:setWidgetInjector(CustomInjector(cosmetic))
+	cosmetic.args.subHeader = cosmetic.args.prefab
+	cosmetic.args.imageText = 'ID: ' .. (cosmetic.args.defindex or 'N/A')
+
+	return mw.html.create():node(cosmetic:createInfobox()):node(cosmetic:_createIntroText())
+end
+
+-- TODO: Store LPDB
+
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:parse(id, widgets)
+	local args = self.caller.args
+
+	if id == 'custom' then
+		local slotText = ((args.slot == 'Persona' and '[[Persona]]') or (args.slot and String.upperCaseFirst(args.slot)))
+		Array.appendWith(widgets,
+			Cell{
+				options = {columns = args.hero and 3 or 10, surpressColon = true},
+				name = args.hero and Template.expandTemplate(mw.getCurrentFrame(), 'Hero entry', {args.hero}) or ' ',
+				content = {
+					'<b>Rarity:</b> ' .. Template.safeExpand(mw.getCurrentFrame(), 'Raritylink', {args.rarity}),
```

Comment:
do we maybe want to get rid of the template?
(and move the colors it sets to css classes?

Reply:
In the future, currently keeping the template calls (goes for all 3)